src/System.Private.CoreLib/src/System/Runtime/MemoryFailPoint.CoreCLR.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.ConstrainedExecution;
+
+namespace System.Runtime
+{
+    public sealed partial class MemoryFailPoint : CriticalFinalizerObject, IDisposable
+    {
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern void GetMemorySettings(out ulong maxGCSegmentSize, out ulong topOfMemory);
```

Comment:
I will modify this to do the same thing as CoreRT and get rid of the file...

Reply:
I added the `GetSegmentSize` FCall to `GC` class. It is named `RuntimeImports.RhGetGCSegmentSize` in CoreRT. I can change it if desired, but it felt like more appropriate place and it should be easy to "forward" in CoreRT.

Reply:
>  `GetSegmentSize` FCall to GC class

This looks good.